src/control/server/storage/bdev/backend_class_test.go

```diff
@@ -120,60 +145,35 @@ func TestBackend_writeJSONFile(t *testing.T) {
 					DeviceList: storage.MustNewBdevDeviceList(test.MockPCIAddrs(1)...),
 				},
 			}).WithStorageEnableHotplug(true),
-			expOut: `
-{
-  "daos_data": {
-    "config": [
-      {
-        "params": {
-          "begin": 0,
```

Comment:
Is there a test verifying the JSON parameters? The lack of that is the only downside I could see to checking the format to the config. Checking the JSON output as a string (even for just a single happy path case) is a good sanity check for our structs' json fields being defined properly when, for example, we change the name of an underlying field like this.

Reply:
I removed the JSON strings in favor of comparing the configuration structs as a way to test `ReadConfig()` and `UnmarshalJSON()`, and also to make these tests a bit more maintainable. Having to repeat the raw JSON strings constantly seems pretty toilsome, and at some point it seems like we're just testing the json package.

I do see your point that it might be useful to have a single test with all of the options so that we can see if some change results in output that wouldn't be parseable by the engine? That having been said, if we update SPDK and there are different config parameters in that new version, we're going to be broken here either way.

Reply:
I'm fine with the general approach for most of the tests. But yeah I think it would be useful to have one unit test that verifies we didn't change the JSON structs unless we meant to. It's significantly easier to debug a Go unit test than a live system with crashing engines.